prow/jira/fakejira/fake.go

```diff
@@ -339,3 +342,197 @@ func (f *FakeClient) UpdateIssue(issue *jira.Issue) (*jira.Issue, error) {
 func (f *FakeClient) UpdateStatus(issueID, statusName string) error {
 	return jiraclient.UpdateStatus(f, issueID, statusName)
 }
+
+type JqlQueries struct {
+	Jql []*JqlQuery
+}
+
+type JqlQuery struct {
+	Key      string
+	Operator string
+	Value    string
+}
+
+var supportedJQLKeys = []string{"project", "category", "id", "issueType", "status"}
+
+var jqlOperators = map[string]string{
+	"equal":    "=",
+	"contains": " IN ",
+}
+
+func contains(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+func trimQuotes(s string) string {
+	if len(s) >= 2 {
+		if c := s[len(s)-1]; s[0] == c && (c == '"' || c == '\'') {
+			return s[1 : len(s)-1]
+		}
+	}
+	return s
+}
+
+func SplitJQLQuery(s string) (JqlQuery, error) {
+	var min int
+	var name string
+	i := 0
+	for operatorName, operatorType := range jqlOperators {
+		ind := strings.Index(s, operatorType)
+		if ind == -1 {
+			continue
+		}
+		if i == 0 {
+			min = ind
+			name = operatorName
+		}
+		if ind < min && ind > -1 {
+			min = ind
+			name = operatorName
+		}
+		i++
+	}
+	if contains(supportedJQLKeys, s[:min]) {
+		return JqlQuery{
+			Key:      s[:min],
+			Operator: name,
+			Value:    trimQuotes(s[min+len(jqlOperators[name]):]),
+		}, nil
+	} else {
+		return JqlQuery{}, fmt.Errorf("failed to split the jql string: %s", s)
+	}
+
+}
+
+func jqlContains(s string, str string) bool {
+	if s[0:1] != "(" && s[len(s)-1:len(s)] != ")" {
+		klog.Warningf("incorrect format of the JQL query: %s", s)
+		return false
+	}
+	a := strings.Split(s[1:len(s)-1], ",")
+	for _, v := range a {
+		if strings.TrimSpace(v) == strings.TrimSpace(str) {
+			return true
+		}
+	}
+	return false
+}
+
+// ParseJQL
+// Supported JQL Operators: "&" (and); JQL fields Operators: "=" (equal), " IN " (contains); JQL Keys: supportedJQLKeys
+func ParseJQL(jql string) JqlQueries {
```

Comment:
I would rather see tests that we send the correct JQL to Jira, rather than attempting to re-implement their query language parser here. This approach is almost certain to differ from the real JQL parser in some way, so we get no more confidence in correctness by doing this as compared to simply a hard equality check against known-good queries.

Reply:
I  agree that attempting to implement the JQL parser would most certainly be different from the real JQL.
But in this case, the Parser supports only a specific type of JQL query. For anything else, we return an error on query validation ( where we check for allowed operators and keys). 
In that sense, these are known-good queries only.